cmake/modules/RootMacros.cmake

```diff
@@ -1338,9 +1338,36 @@ function(ROOT_EXECUTABLE executable)
   if(ARG_TEST) # we are building a test, so add EXCLUDE_FROM_ALL
     set(_all EXCLUDE_FROM_ALL)
   endif()
-  include_directories(BEFORE ${CMAKE_BINARY_DIR}/include)
-  add_executable( ${executable} ${_all} ${exe_srcs})
-  target_link_libraries(${executable} ${ARG_LIBRARIES} )
+  if(NOT (PROJECT_NAME STREQUAL "ROOT"))
+    # only for non-ROOT executable use $ROOTSYS/include
+    include_directories(BEFORE ${CMAKE_BINARY_DIR}/include)
```

Comment:
But if not `ROOT`, is `CMAKE_BINARY_DIR` really `ROOTSYS`?

Reply:
No idea, all logic around `(PROJECT_NAME STREQUAL "ROOT")` not always clear for me.
But again - idea of my changes - adjust code only for ROOT itself